x-pack/plugins/reporting/export_types/common/execute_job/index.test.js

```diff
@@ -0,0 +1,288 @@
+        baz: 'quix',
+      };
+
+      const { conditionalHeaders } = await getConditionalHeaders({ job: { },
+        filteredHeaders: permittedHeaders,
+        server: mockServer });
+
+      expect(conditionalHeaders.conditions.hostname).toEqual(config['server.host']);
+
+    });
+
+    test(`uses port from reporting config if set`, async () => {
+      config['xpack.reporting.kibanaServer.port'] = 443;
+
+      const permittedHeaders = {
+        foo: 'bar',
+        baz: 'quix',
+      };
+
+      const { conditionalHeaders } = await getConditionalHeaders({ job: { },
+        filteredHeaders: permittedHeaders,
+        server: mockServer });
+
+      expect(conditionalHeaders.conditions.port).toEqual(config['xpack.reporting.kibanaServer.port']);
+
+    });
+
+    test(`uses port from server if reporting config not set`, async () => {
+      const permittedHeaders = {
+        foo: 'bar',
+        baz: 'quix',
+      };
+
+      const { conditionalHeaders } = await getConditionalHeaders({ job: { },
+        filteredHeaders: permittedHeaders,
+        server: mockServer });
+
+      expect(conditionalHeaders.conditions.port).toEqual(config['server.port']);
+
+    });
+
+    test(`uses basePath from server config`, async () => {
+      const permittedHeaders = {
+        foo: 'bar',
+        baz: 'quix',
+      };
+
+      const { conditionalHeaders } = await getConditionalHeaders({ job: { },
+        filteredHeaders: permittedHeaders,
+        server: mockServer });
+
+      expect(conditionalHeaders.conditions.basePath).toEqual(config['server.basePath']);
+
+    });
+
+    test(`uses protocol from reporting config if set`, async () => {
+      config['xpack.reporting.kibanaServer.protocol'] = 'https';
+      const permittedHeaders = {
+        foo: 'bar',
+        baz: 'quix',
+      };
+
+      const { conditionalHeaders } = await getConditionalHeaders({ job: { },
+        filteredHeaders: permittedHeaders,
+        server: mockServer });
+
+      expect(conditionalHeaders.conditions.protocol).toEqual(config['xpack.reporting.kibanaServer.protocol']);
+
+    });
+
+    test(`uses protocol from server.info`, async () => {
+      const permittedHeaders = {
+        foo: 'bar',
+        baz: 'quix',
+      };
+
+      const { conditionalHeaders } = await getConditionalHeaders({ job: { },
+        filteredHeaders: permittedHeaders,
+        server: mockServer });
+
+      expect(conditionalHeaders.conditions.protocol).toEqual(mockServer.info.protocol);
+
+    });
+
+    test(`adds forceNow to hash's query, if it exists`, async () => {
+
+      const forceNow = '2000-01-01T00:00:00.000Z';
+
+      const { urls } = await addForceNowQuerystring({ job: { relativeUrl: '/app/kibana#/something', forceNow }, server: mockServer });
+
+      expect(urls[0]).toEqual('http://localhost:5601/sbp/app/kibana#/something?forceNow=2000-01-01T00%3A00%3A00.000Z');
+
+    });
+
+    test(`appends forceNow to hash's query, if it exists`, async () => {
```

Comment:
What about creating a `add_force_now_query_string.test.ts` and pushing this test and the above into there?  Since all it's doing is testing that one function, seems like a good candidate to pull out.